### PR TITLE
Augments the add database command to handle zip files

### PR DESCRIPTION
**Pull request by aeisenberg**
The add database command can now add databases by zip file. When a
file is selected, the zip file is attempted to be extracted into a
directory managed by the extension.

Once extracted, a database is searched for, by looking for a .dbinfo
file.

Crucially, we are using the same infrastructure to download a database
as we are to add a database by zip file.

## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/master/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [x] `@github/product-docs-dsp` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: extensions/ql-vscode/src/databaseFetcher.ts [hunk 1]
```diff
@@ -45,36 +45,27 @@ async function databaseFetcher(
...
-  // if there is a single directory inside, then assume that's what we want to import
-  const dirs = await fs.readdir(unzipPath);
-  const dbPath = dirs?.length === 1 && (await fs.stat(path.join(unzipPath, dirs[0]))).isDirectory
-    ? path.join(unzipPath, dirs[0])
-    : unzipPath;
-
-  // might need to upgrade before importing...
-  const item = await databasesManager.openDatabase(Uri.parse(dbPath));
-  databasesManager.setCurrentDatabaseItem(item);
+  const dbPath = await findDirWithFile(unzipPath, '.dbinfo');
```
Comment (adityasharad) [inline extensions/ql-vscode/src/databaseFetcher.ts:60]: Note that this is `codeql-database.yml` for newer databases.

Comment (aeisenberg) [reply extensions/ql-vscode/src/databaseFetcher.ts:60]: Yikes. Is there a stable file to look for?

Comment (jcreedcmu) [reply extensions/ql-vscode/src/databaseFetcher.ts]: Seems like we want to check for the presence of either `codeql-database.yml` or `.dbinfo`, yeah?

Comment (jcreedcmu) [reply extensions/ql-vscode/src/databaseFetcher.ts]: Checking for the presence of either of those two seems appropriate to me.

**Comment by jcreedcmu** [comment]
Unfortunately, this implementation doesn't work for me on Linux, and I suspect it won't work on Windows, based on docs in https://code.visualstudio.com/api/references/vscode-api#OpenDialogOptions --- the dialog lets me select only folders, not files, if `canSelectFolders` is set, even if `canSelectFiles` is true. We might have to rethink how this works.

On alternative might be to enable the 'set current database' context menu command in the vscode file explorer for .zip files. I think this would work well in the case of codespaces with snapshots you downloaded locally, since then the flow could be just drag-and-drop, right-click and set as current database.

**Comment by jcreedcmu** [comment]
I think doing something like
```
        {
          "command": "codeQL.setCurrentDatabase",
          "group": "9_qlCommands",
          "when": "resourceScheme == codeql-zip-archive || explorerResourceIsFolder || resourceExtname == .zip"
        },
```
will work.

**Comment by aeisenberg** [comment]
That's interesting. I will try your suggestion out.

**Comment by aeisenberg** [comment]
Ping @github/product-docs-dsp! 

Some new features in this PR

* We've changed the icons on the databases view
* We've added two new commands on the databases view: 
  - Choose Database from Internet (the cloud icon), which is the same action I implemented in #358.
  - Choose Database from Archive (the file icon), which will take a zip file, and unzip it into a extension-controlled location and import the database into the workspace.
  - Choose Database from Folder (the folder icon), which is existing functionality to add a database from a folder, except that the icon has changed.

**Comment by aeisenberg** [comment]
@jcreedcmu, I'll do your suggestion in a separate PR.

**Comment by aeisenberg** [comment]
Fixes #369.

**Pull request merged**